plugins/dialog/plugin.js

```diff
@@ -3001,7 +3006,13 @@ CKEDITOR.DIALOG_STATE_BUSY = 2;
 
 	CKEDITOR.dialogCommand.prototype = {
 		exec: function( editor ) {
-			editor.openDialog( this.dialogName );
+			var that = this;
+			editor.openDialog( this.dialogName, function( dialog ) {
+				// Select different tab if it's provided (#830).
+				if ( that && that.tab ) {
```

Comment:
Is `that.tab` available only here (or can it be changed between `exec` command is executed and this line)? If not instead of using `var that = this`, you may already assign the `tab` value there.